bytecomp/matching.ml

```diff
@@ -1604,8 +1613,14 @@ let get_args_record num_fields p rem = match p with
 
 let matcher_record num_fields p rem = match p.pat_desc with
 | Tpat_or (_,_,_) -> raise OrPat
-| Tpat_var _      -> get_args_record num_fields omega rem
-| _               -> get_args_record num_fields p rem
+| Tpat_any
+| Tpat_var _      ->
+  record_matching_line num_fields [] @ rem
+| Tpat_record ([], _) when num_fields = 0 -> rem
```

Comment:
Interesting, how did that line about the empty record end up here?

Reply:
It seemed better to be robust rather than `assert false` (especially considering other GPRs/discussions currently happening).
As we've discovered, `assert false`s can go unnoticed for a while even when the underlying invariant cannot be safely assumed anymore.

Reply:
On the other hand, the module code generally assumes that record are non-empty everywhere (I thought about this when writing my part of the patch, notice the ubiquitous `(_, lbl, _) :: _` patterns). I think it would have been better, for consistency, to have all the code share this same assumption, rather than have some parts be clever and some not. But I also think that the new line you added is correct, and will not require further work by people willing to add empty records, so it's fine to leave it in.